consumer-test/consumerTestController.groovy

```diff
@@ -0,0 +1,144 @@
+import static ConsumerTestUtils.exitPrematurely
+import static ConsumerTestUtils.listYamlInDirRecursive
+import static ConsumerTestUtils.newEmptyDir
+import static ConsumerTestUtils.notifyGithub
+
+AUXILIARY_SLEEP_MS = 10000
+// Build is killed at 50 min, print log to console at minute 45
+PRINT_LOGS_AFTER_45_MINUTES_COUNTDOWN = (45 * 60 * 1000) / AUXILIARY_SLEEP_MS
+WORKSPACES_ROOT = 'workspaces'
+TEST_CASES_DIR = 'testCases'
+
+EXCLUDED_FROM_CONSUMER_TESTING_REGEXES = [
+    /^documentation\/.*$/,
+    /^.travis.yml$/,
+    /^test\/.*$/
+]
+
+newEmptyDir(WORKSPACES_ROOT)
+ConsumerTestUtils.workspacesRootDir = WORKSPACES_ROOT
+ConsumerTestUtils.libraryVersionUnderTest = "git log --format=%H -n 1".execute().text.trim()
+ConsumerTestUtils.repositoryUnderTest = System.getenv('TRAVIS_REPO_SLUG') ?: 'SAP/jenkins-library'
+
+def testCaseThreads
+def filePath = args?.find()
+if (filePath) {
```

Comment:
Why is `filePath` a good property for making the decision if we run locally? Why no explict flag for this `--run-local`?